packages/gateway-conformance/src/fixtures/basic-server.ts

```diff
@@ -13,9 +25,165 @@ const log = logger('basic-server')
+async function callVerifiedFetch (req: IncomingMessage, res: Response, { serverPort, useSessions, verifiedFetch }: CallVerifiedFetchOptions): Promise<void> {
+  const log = logger('basic-server:request')
+  if (req.method === 'OPTIONS') {
+    res.writeHead(200)
+    res.end()
+    return
+  }
+  if (req.method === 'HEAD') {
+    res.writeHead(200)
+    res.end()
+    return
+  }
+
+  if (req.url == null) {
+    // this should never happen
+    log.error('No URL provided, returning 400 Bad Request')
+    res.writeHead(400)
+    res.end('Bad Request')
+    return
+  }
+
+  // @see https://github.com/ipfs/gateway-conformance/issues/185#issuecomment-2123708150
+  let fixingGwcAnnoyance = false
+  if (req.headers.host != null && req.headers.host === 'localhost') {
+    log.trace('set fixingGwcAnnoyance to true for %s', new URL(req.url, `http://${req.headers.host}`).href)
+    fixingGwcAnnoyance = true
+    req.headers.host = `localhost:${serverPort}`
+  }
+
+  const fullUrlHref = new URL(req.url, `http://${req.headers.host}`)
+
+  const urlLog = logger(`basic-server:request:${fullUrlHref}`)
+  urlLog('configuring request')
+  urlLog.trace('req.headers: %O', req.headers)
+  let requestController: AbortController | null = new AbortController()
+  // we need to abort the request if the client disconnects
+  const onReqEnd = (): void => {
+    urlLog('client disconnected, aborting request')
+    requestController?.abort()
+  }
+  req.on('end', onReqEnd)
+
+  const reqTimeout = setTimeout(() => {
+    /**
+     * Abort the request because it's taking too long.
+     * This is only needed for when @helia/verified-fetch is not correctly
+     * handling a request and should not be needed once we have 100% gateway
+     * conformance coverage.
+     */
+    urlLog.error('timing out request')
+    requestController?.abort()
+  }, 2000)
+  reqTimeout.unref() // don't keep the process alive just for this timeout
+
+  const onResFinish = (): void => {
+    urlLog.trace('response finished, aborting signal')
+    requestController?.abort()
+  }
+  res.on('finish', onResFinish)
+
+  try {
+    urlLog.trace('calling verified-fetch')
+    const resp = await verifiedFetch(fullUrlHref.toString(), { redirect: 'manual', signal: requestController.signal, session: useSessions, allowInsecure: true, allowLocal: true, headers: convertNodeJsHeadersToFetchHeaders(req.headers) })
+    urlLog.trace('verified-fetch response status: %d', resp.status)
+
+    const headers = convertFetchHeadersToNodeJsHeaders({ resp, log: urlLog, fixingGwcAnnoyance, serverPort })
+
+    res.writeHead(resp.status, headers)
+    if (resp.body == null) {
+      // need to convert ArrayBuffer to Buffer or Uint8Array
+      res.write(Buffer.from(await resp.arrayBuffer()))
+      urlLog.trace('wrote response')
+    } else {
+      // read the body of the response and write it to the response from the server
+      const reader = resp.body.getReader()
```

Comment:
Can you do something like this:
```suggestion
      const reader = resp.body.getReader()
      reader.pipe(res)
```

Reply:
Unfortunately, fetch streams/readable and webstreams arent directly compatible. I think alex has a package that handles this though.. will look into it

Reply:
Good to know. Thanks for sharing